### ISSUE TITLE
RangeError: Maximum call stack size exceeded (facing issue in prod build alone, dev version working fine)

### ISSUE THREAD

**Issue by Arun12Muralidharan** [vercel/next.js#59075]
### Link to the code that reproduces this issue

https://github.com/Arun12Muralidharan/BlogApp

### To Reproduce

1. After build in local / vercel site, while trying to login user / register user / create new post / edit existing post. Facing "RangeError: Maximum call stack size exceeded". 
2. Page got stuck there itself (to overcome such behaviour, i had added error page), still same issue.
3. The issue not occuring in dev mode.

Please checkout the running version (with issue) of the app at
"https://am-blog-app.vercel.app/"

### Current vs. Expected behavior

Current Behaviour:
RangeError: Maximum call stack size exceeded in prod version alone. (In Dev mode, no issues).

Expected:
Error free module in prod version too.

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 10 Pro Service Pack 1
Binaries:
  Node: 18.17.1
  npm: N/A
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 13.5.6
  eslint-config-next: 13.5.6
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.2.2
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Not sure

### Additional context

Please checkout the running version (with issue) of the app at
"https://am-blog-app.vercel.app/"

**Comment by JacobKee** [comment]
Hi,

Have you solved the issue above?
I am facing this issue as well.
RangeError: Maximum call stack size exceeded
    at handleErrorFromBinding (node:internal/fs/utils:351:17)
    at Object.mkdirSync (node:fs:1379:3)

**Comment by Arun12Muralidharan** [comment]
Hi JacobKee, 

Issue not resolved yet, I believe it might be a bug in next / react integration.

**Comment by isaacrowntree** [comment]
I have the same issue here.

**Comment by thevgseven** [comment]
`pnpm run build` - executed successfully locally, but on vercel build execution failed with these error: `RangeError: Maximum call stack size exceeded`

![build-failed](https://github.com/vercel/next.js/assets/81641747/a4f22a54-222e-4d64-a0e7-56740987b0d9)

Any Solution ?

**Comment by JacobKee** [comment]
> `pnpm run build` - executed successfully locally, but on vercel build execution failed with these error: `RangeError: Maximum call stack size exceeded`
> 
> ![build-failed](https://private-user-images.githubusercontent.com/81641747/296443552-a4f22a54-222e-4d64-a0e7-56740987b0d9.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUxNDAzMTYsIm5iZiI6MTcwNTE0MDAxNiwicGF0aCI6Ii84MTY0MTc0Ny8yOTY0NDM1NTItYTRmMjJhNTQtMjIyZS00ZDY0LWEwZTctNTY3NDA5ODdiMGQ5LlBORz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTEzVDEwMDAxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBmMjJhNDY3NjQ4NzNmY2JlNzY1NTMzZTczMTYwODFjMGVjZDRlZTRiNjAwNmFjMWM2ZTY3YWFkNjZjMzcwNWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.G0AUEfVeQG9jd3vHn4MBxknTzcP_8n1aY2NKosoDTSk)
> 
> Any Solution ?

We found out that it is the log plugin issue.
It is not supported in vercel.
After we removed the log4js, it is working fine now

**Comment by thevgseven** [comment]
> > `pnpm run build` - executed successfully locally, but on vercel build execution failed with these error: `RangeError: Maximum call stack size exceeded`
> > ![build-failed](https://private-user-images.githubusercontent.com/81641747/296443552-a4f22a54-222e-4d64-a0e7-56740987b0d9.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUxNDAzMTYsIm5iZiI6MTcwNTE0MDAxNiwicGF0aCI6Ii84MTY0MTc0Ny8yOTY0NDM1NTItYTRmMjJhNTQtMjIyZS00ZDY0LWEwZTctNTY3NDA5ODdiMGQ5LlBORz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTEzVDEwMDAxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBmMjJhNDY3NjQ4NzNmY2JlNzY1NTMzZTczMTYwODFjMGVjZDRlZTRiNjAwNmFjMWM2ZTY3YWFkNjZjMzcwNWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.G0AUEfVeQG9jd3vHn4MBxknTzcP_8n1aY2NKosoDTSk)
> > Any Solution ?
> 
> We found out that it is the log plugin issue. It is not supported in vercel. After we removed the log4js, it is working fine now

How to remove log4js ?

**Comment by thevgseven** [comment]
Error Resolved Successfully, error counter by `sharp` lib next use for Image Processing to serve images and by reducing the version of sharp `0.33.2` to `0.33.1` resolves the error and `vercel build` executed successfully...

**Comment by faradaytrs** [comment]
> Error Resolved Successfully, error counter by `sharp` lib next use for Image Processing to serve images and by reducing the version of sharp `0.33.2` to `0.33.1` resolves the error and `vercel build` executed successfully...

that is not a solution, it's just a workaround.

**Comment by Arun12Muralidharan** [comment]
The issue is gets resolved in nextjs v14. I had craeted the issue, when i was using v13

**Comment by faradaytrs** [comment]
no it actually exists now i think @Arun12Muralidharan

**Comment by tapnisu** [comment]
> The issue is gets resolved in nextjs v14. I had craeted the issue, when i was using v13

Still exists for me with sharp `0.33.2` and next `14.0.4`

**Comment by jeffreys-cat** [comment]
> > The issue is gets resolved in nextjs v14. I had craeted the issue, when i was using v13
> 
> Still exists for me with sharp `0.33.2` and next `14.0.4`

`0.33.1` and next `14.0.4` not work

**Comment by thevgseven** [comment]
> > > The issue is gets resolved in nextjs v14. I had craeted the issue, when i was using v13
> > 
> > 
> > Still exists for me with sharp `0.33.2` and next `14.0.4`
> 
> `0.33.1` and next `14.0.4` not work

 `0.33.1` and next `14.0.4` works

**Comment by thevgseven** [comment]
I have open the [issue](https://github.com/lovell/sharp/issues/3944) in `sharp` lib but @lovell says it's not the issue of `sharp` lib it's the issue with lib `micromatch` or `next.js 14.0.4`.

**Comment by FleetAdmiralJakob** [comment]
I tried out all different sharp versions. It is not a problem of sharp it is a problem of Next.js.

**Comment by leerob** [comment]
P.S. if you are deploying to Vercel, you do not need to install `sharp` 👍

**Comment by leerob** [comment]
Fix for this is in `v14.0.5-canary.60` https://github.com/vercel/next.js/pull/60740

**Comment by FleetAdmiralJakob** [comment]
@leerob I never intended to install sharp but Next.js throw an terminal warn that I should install sharp for production. Maybe this warn should be deleted.
